app/js/enhancedDate.js

```diff
@@ -1,5 +1,76 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var enhancedDate = (function(window, undefined) {
 
+	var _date;
+	var weekday = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
+	var month = ["January", "February", "March", "April", "May", "June", "July", "Auguest", "September", "October", "Novenber", "December"];
+	var current = new Date();
+	var setDate = false;
+
+	function _setDate(oDate) {
+		if(oDate instanceof Date) {
+			_date = oDate;
+		} else {
+			_date = current;
+		}
+
+		setDate = true;
+	}
+	function _getDate(oDate) {
+		if(!setDate) {
+			_setDate();
+		}
+
+		if(oDate === true) {
+			return _date;
+		} else {
+			return _date.getTime();
+		}
+	}
+	function _getDayName() {
+		if(!setDate) {
+			_setDate();
+		}
+		return weekday[_date.getDay()];
+	}
+	function _getMonthName() {
+		if(!setDate) {
+			_setDate();
+		}
+		return month[_date.getMonth()];
+	}
+	function _isFuture() {
+		if(!setDate) {
+			_setDate();
+		}
+		if( _date.getDay() >= current.getDay() ) {
```

Comment:
Calling getDay doesn't provide enough information to make this comparison. The getDay method returns a value from 0 to 6 based on the day of the week. What will work here instead is getTime().